execution_engine2.spec

```diff
@@ -190,43 +216,120 @@ module execution_engine2 {
     funcdef start_job(StartJobParams params) returns () authentication required;
 
     /*
-        projection: projecct certain fields to return. default None.
+        projection: project certain fields to return. default None.
+        projection strings can be one of:
+            ...
     */
     typedef structure {
         job_id job_id;
         list<string> projection;
     } CheckJobParams;
 
+    /*
+        job_id - string - id of the job
+        user - string - user who started the job
+        wsid - int - id of the workspace where the job is bound
+        authstrat - string - what strategy used to authenticate the job
+        job_input - object - inputs to the job (from the run_job call)  ## TODO - verify
+        updated - string - timestamp of the last time the status was updated
+        running - string - timestamp of when it entered the running state
+        created - string - timestamp when the job was created
+        finished - string - timestamp when the job was finished
+        status - string - status of the job. one of the following:
+            created - job has been created in the service
+            estimating - an estimation job is running to estimate resources required for the main
+                         job, and which queue should be used
+            queued - job is queued to be run
+            running - job is running on a worker node
+            finished - job was completed successfully
+            error - job is no longer running, but failed with an error
+            terminated - job is no longer running, terminated either due to user cancellation,
+                         admin cancellation, or some automated task
+        error_code - int - internal reason why the job is an error. one of the following:
+            0 - unknown
+            1 - job crashed
+            2 - job terminated by automation
+            3 - job ran over time limit
+            4 - job was missing its automated output document
+            5 - job authentication token expired
+        errormsg - string - message (e.g. stacktrace) accompanying an errored job
+
+        terminated_code - int - internal reason why a job was terminated, one of:
+            0 - user cancellation
+            1 - admin cancellation
+            2 - terminated by some automatic process
+
+        @optional error_code
+        @optional errormsg
+        @optional terminated_code
+        @optional estimating
+        @optional running
+        @optional finished
+    */
+
+
+    typedef structure {
+        job_id _id;
+        string user;
+        string authstrat;
+        int wsid;
+        string status;
+        RunJobParams job_input;
+        timestamp created;
+        timestamp updated;
+
+        timestamp estimating
+        timestamp running;
+        timestamp finished;
+
+        int error_code;
+        string errormsg;
+
```

Comment:
I think we also return an "error" object as well.

Reply:
Fixed. Just used the JsonRpcError object that's there. It has the same structure as the model.